conda_forge_tick/update_recipe/version.py

```diff
@@ -527,3 +529,28 @@ def update_version(raw_meta_yaml, version, hash_type="sha256"):
     else:
         logger.critical("Recipe did not change in version migration!")
         return None, errors
+
+
+def update_recipe_yaml_version(
+    raw_recipe_yaml: str, version: str, hash_type: str = "sha256"
+) -> tuple[str | None, set[str]]:
+    """Update the version in a recipe.
+
+    Parameters
+    ----------
+    raw_recipe_yaml : str
+        The recipe meta.yaml as a string.
+    version : str
+        The version of the recipe.
+    hash_type : str, optional
+        The kind of hash used on the source. Default is sha256.
+
+    Returns
+    -------
+    updated_recipe_yaml : str or None
+        The updated meta.yaml. Will be None if there is an error.
+    errors : set of str
+        A set of strings giving any errors found when updating the
+        version. The set will be empty if there were no errors.
+    """
+    raise NotImplementedError()
```

Comment:
This still needs to be implemented.
`update_version` the `meta.yaml` pendant contains *a lot* of logic. One has to have a close look which parts of it are not necessary for `recipe.yaml` and how to adapt the rest that is.